CONTRIBUTING.md

```diff
@@ -1,32 +1,92 @@
+  - cargo                $ pip3 install --editable .                           # Building wheels for collected packages: janitor, aiohttp-apispec, breezy -> Building editable for janitor (pyproject.toml) ... error -> distutils.errors.DistutilsPlatformError: can't find Rust compiler
+  - gcc                  $ pip3 install --editable .                           # Building wheel for breezy (pyproject.toml) ... error -> error: command 'x86_64-linux-gnu-gcc' failed: No such file or directory
+  - libpython3-dev       $ pip3 install --editable .                           # Building wheel for breezy (pyproject.toml) ... error -> breezy/bzr/_simple_set_pyx.c:32:10: fatal error: Python.h: No such file or directory
+  - libssl-dev           $ pip3 install --editable .                           # warning: openssl-sys@0.9.105: Could not find directory of OpenSSL installation, and this `-sys` crate cannot proceed without this knowledge. If OpenSSL is installed and this crate had trouble finding it,  you can set the `OPENSSL_DIR` environment variable for the compilation process. See stderr section below for further information.
+  - g++                  $ pip3 install --editable .                           # error: linking with `cc` failed: exit status: 1 -> distutils.errors.CompileError: `cargo build --manifest-path worker/Cargo.toml --message-format=json-render-diagnostics -v --features 'cli debian'` failed with code 101
+  - swig                 $ pip3 install --editable .                           # Running setup.py install for gpg ... error -> Using gpgme.h from /usr/include/gpgme.h -> error: command 'swig' failed: No such file or directory
+
+- Using python3-gpg over PyPi GPG as having issues with Debian Testing (okay with Debian 12)
+  https://github.com/jelmer/janitor/issues/931
+
+Dependencies
+  - cargo -> rustc
+-->
+
+- - -
+
+In addition to these packages, will need to use Python's PIP and a virtual
+environment to install the rest of the Python-based dependencies:
 
 ```console
+$ sudo apt install \
+    python3-venv
 $ git clone https://github.com/jelmer/janitor.git
 $ cd janitor/
-$ python3 -m venv
-$ . ./bin/activate
-$ pip3 install -e .[dev,debian]
+$ python3 -m venv .venv
+$ . ./.venv/bin/activate
+$ pip3 install --editable .[dev]
+```
+<!--
+Via python3-venv, there will be: `./.venv/bin/pip` (which is why don't need python3-pip)
+-->
+
+_Python's package management over OS network apt package, as they may be too dated._
+
+### Production Environment
+
+We would recommend using containers to to run each of the Janitor services.
+
+There are (daily) [pre-built containers](https://github.com/jelmer?tab=packages&repo_name=janitor),
+otherwise you can create them yourself:
+
+```console
+$ sudo apt install \
+    buildah \
+    make
+$ make build-all
+```
+
+- - -
+
+Running the containers can be done however is best to suite your environment,
```

Comment:
This should probably now be in the other PR?

Reply:
Dropped for the time being